### ISSUE TITLE
Issue with Hiero font generator on Win10

### ISSUE THREAD

**Issue by EdwardAngeles** [GoranM/bdx#571]
When I enable "Show Framerate and Profile", I get this error:

`FileNotFoundError: [Errno 2] No such file or directory: 'E:\\bdx\\projects\\FirstProject\\android\\assets\\bdx\\fonts\\BDXFontMono.fntx'`

I checked the location and indeed there aren't any fonts there. I'm using the last version (0.2.3).

**Comment by SolarLune** [comment]
Hey, there!

Is this the default demo scene? 
Does it run fine otherwise? 
What OS are you using?

**Comment by EdwardAngeles** [comment]
- Yes, it's the default demo scene.
- Since the file (the .fntx file) is messing. I don't think there it's something I can do.
- Windows 10.

I just created a new bdx project, enable the profile and it didn't work. Oh, just tried with the developer version, same thing.

**Comment by SolarLune** [comment]
Hm, that's strange...

Something you might want to try, especially if you've never used BDX before, is running it without the profiler enabled. Does it work?  If so, then try turning on the profiler.

If that doesn't work, as a quick fix, [here's](https://www.dropbox.com/s/61rfemn30fqwyts/BDXFontMono.fntx?dl=0) the missing fntx file so you can at least continue using BDX.

@rafcolson might be able to help out a bit more, as he knows the most about the profiler.

**Comment by EdwardAngeles** [comment]
It runs without the profiler.

I use your file, it fixed that problem but other one shows:

`File not found: bdx\textures\__FNT_BDXFontMono.png`

Obviously, I'm messing that one two. Would you mind uploading it too? Hopefully it's the last one left.

**Comment by rafcolson** [comment]
Try this. In the assets/scenes folder remove Profiler.bdx. And run the game again.

If that doesn't work remove the project and start a new one.

It sure would be strange if you are missing the fntx and png after a first run with the profiler enabled. Let me know what happens.

**Comment by EdwardAngeles** [comment]
Thanks for the quick answers.

- I don't see any __Profiler.bdx on assets\bdx\scenes, just Scene.bdx. I created a new project: same thing.

- I can run a project normally until I enable "Show Fremarate and Profile". After that I start to get the error, even if I disable it, it doesn't run.

- When I run BDX with the profiler for the first time no popups are show.

**Comment by rafcolson** [comment]
>I don't see any __Profiler.bdx on assets\bdx\scenes, just Scene.bdx. I created a new project: same thing.

That's not normal. With a fresh project and framerate and profile enabled this ```__Profiler.bdx``` should be added to ```assets\bdx\scenes```.

>When I run BDX with the profiler for the first time no popups are show.

Also this is not normal. With a fresh project and framerate and profile enabled a small window "Hiero Font generator" should appear briefly. If it doesn't appear it means that the ```*.fntx``` and ```*.png``` are also not created.

How about doing another test. Start a fresh project, don't enable framerate and profile, but just add a text object to the scene. Run BDX.

Some other question. What Blender version are you using?

**Comment by rafcolson** [comment]
>Obviously, I'm messing that one too. Would you mind uploading it too? Hopefully it's the last one left.

Forgot about that. Sorry. Here it is: [__FNT_BDXFontMono.png](https://www.dropbox.com/s/g1a8kumi7j2hudl/__FNT_BDXFontMono.png?dl=0).

**Comment by EdwardAngeles** [comment]
Ok, this is what I got:

- Added some text to the scene:
`FileNotFoundError: [Errno 2] No such file or directory: 'E:\\bdx\\projects\\TestProfile3\\android\\assets\\bdx\\fonts\\Bfont.fntx'`

-Blender version 2.76b.

By the way: looking at the console. I've seen something saying that a window wasn't supported. I can't see it know, don't know. It might be the Hiero Font generator?

**Comment by rafcolson** [comment]
>Added some text to the scene: ```FileNotFoundError: [Errno 2] No such file or directory: 'E:\\bdx\\projects\\TestProfile3\\android\\assets\\bdx\\fonts\\Bfont.fntx'```

Thanks for testing. Yes, it seems that there's a problem with the font generator.

>I've seen something saying that a window wasn't supported.

It probably would help to share the message from the console. Maybe that will point us into the right direction, trying to tackle this issue. Thanks for you help.

**Comment by EdwardAngeles** [comment]
The error about the window isn't appearing.

Anyways, I'll leave the Exception/Error I'm getting:

`Exception in thread "LWJGL Application" com.badlogic.gdx.utils.GdxRuntimeException: OpenGL is not supported by the video driver.
        at com.badlogic.gdx.backends.lwjgl.LwjglGraphics.createDisplayPixelFormat(LwjglGraphics.java:299)
        at com.badlogic.gdx.backends.lwjgl.LwjglGraphics.setupDisplay(LwjglGraphics.java:179)
        at com.badlogic.gdx.backends.lwjgl.LwjglApplication.mainLoop(LwjglApplication.java:141)
        at com.badlogic.gdx.backends.lwjgl.LwjglApplication$1.run(LwjglApplication.java:123)
Caused by: org.lwjgl.LWJGLException: Pixel format not accelerated
        at org.lwjgl.opengl.WindowsPeerInfo.nChoosePixelFormat(Native Method)
        at org.lwjgl.opengl.WindowsPeerInfo.choosePixelFormat(WindowsPeerInfo.java:52)
        at org.lwjgl.opengl.WindowsDisplay.createWindow(WindowsDisplay.java:253)
        at org.lwjgl.opengl.Display.createWindow(Display.java:306)
        at org.lwjgl.opengl.Display.create(Display.java:848)
        at org.lwjgl.opengl.Display.create(Display.java:757)
        at com.badlogic.gdx.backends.lwjgl.LwjglGraphics.createDisplayPixelFormat(LwjglGraphics.java:289)
        ... 3 more
Traceback (most recent call last):
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 852, in execute
    return export(context, self.filepath, self.scene_name, self.exprun, self.apply_modifier)
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 804, in export
    bdx["models"].update(srl_models_text(ts, fonts_dir))
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 246, in srl_models_text
    for t in texts}
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 246, in <dictcomp>
    for t in texts}
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 234, in fntx
    with open(j(fntx_dir, t.font.name + ".fntx"), 'r') as f:
FileNotFoundError: [Errno 2] No such file or directory: 'E:\\bdx\\projects\\Test2\\android\\assets\\bdx\\fonts\\BDXFontMono.fntx'

location: C:\Program Files\Blender Foundation\Blender\2.76\scripts\modules\bpy\ops.py:189

location: C:\Program Files\Blender Foundation\Blender\2.76\scripts\modules\bpy\ops.py:189
Error: Traceback (most recent call last):
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 852, in execute
    return export(context, self.filepath, self.scene_name, self.exprun, self.apply_modifier)
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 804, in export
    bdx["models"].update(srl_models_text(ts, fonts_dir))
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 246, in srl_models_text
    for t in texts}
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 246, in <dictcomp>
    for t in texts}
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 234, in fntx
    with open(j(fntx_dir, t.font.name + ".fntx"), 'r') as f:
FileNotFoundError: [Errno 2] No such file or directory: 'E:\\bdx\\projects\Traceback (most recent call last):
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\ops\exprun.py", line 52, in execute
    bpy.ops.export_scene.bdx(filepath=file_path, scene_name=scene.name, exprun=True)
  File "C:\Program Files\Blender Foundation\Blender\2.76\scripts\modules\bpy\ops.py", line 189, in __call__
    ret = op_call(self.idname_py(), None, kw)
RuntimeError: Error: Traceback (most recent call last):
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 852, in execute
    return export(context, self.filepath, self.scene_name, self.exprun, self.apply_modifier)
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 804, in export
    bdx["models"].update(srl_models_text(ts, fonts_dir))
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 246, in srl_models_text
    for t in texts}
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 246, in <dictcomp>
    for t in texts}
  File "C:\Users\edwar\AppData\Roaming\Blender Foundation\Blender\2.76\scripts\addons\bdx\exporter.py", line 234, in fntx
    with open(j(fntx_dir, t.font.name + ".fntx"), 'r') as f:
FileNotFoundError: [Errno 2] No such file or directory: 'E:\\bdx\\projects\\Test2\\android\\assets\\bdx\\fonts\\BDXFontMono.fntx'

location: C:\Program Files\Blender Foundation\Blender\2.76\scripts\modules\bpy\ops.py:189`

**Comment by EdwardAngeles** [comment]
I got it working! It just a walk around, thought:

- I Added bdx\fonts\BDXFontMono.fntx
- It threw an error: BDXFontMono.001.fntx was missing.
- I Copied the original on the same folder and I named BDXFontMono.001.fntx.
- Then, it was looking for the texture. Error: "File not found: bdx\textures\__FNT_BDXFontMono.png"
- I added that file.
- It worked.

In a nutshell:
- bdx\fonts\BDXFontMono.fntx
- bdx\fonts\BDXFontMono.001.fntx
- bdx\textures\__FNT_BDXFontMono.png

**Comment by rafcolson** [comment]
>After adding those files, enabling profile and running bdx, it created the __Profiler.bdx file.

Good that this workaround fixes the problem for now. The source of the issue however remains. When you want to add text, you'll get into problems. So you may want to leave this issue open until it is resolved.

>Quick question: Is there a way to change how often the profile window is updated. Those number changing that fast annoys me.

No, the profile is updated with every logic tic. For more info on the profiler and the options it has you can have a look [here](https://github.com/GoranM/bdx/wiki/API-Profiler).

One other thing. Because this issue is not so much related to the profiler, but to the Hiero font generator, it would help to rename the issue: "Issue with Hiero font generator on Win10" or something like that. Thanks again for the posting the Error.

**Comment by EdwardAngeles** [comment]
Well, I found out the problem. There was a problem with my laptop video driver and windows, the laptop is kind of old. I install ubutun on it and Boom! It's working perfectly!

**Comment by SolarLune** [comment]
Hmm, I see. Well, I guess you can close this issue. Glad it all worked out!

**Comment by billzo** [comment]
I think it might be worth looking into this again.. I'm getting the exact same error, and I'm running on Linux Mint 17.3 Rosa. It happens like above, only whenever I start the profiler.

(Well, first, it was telling me that "bpy.ops.wm.append" could no be found within the exprun.py file... I assume there are some compatibility issues with Blender scripts re: this atm, because when I changed that to the _deprecated_ "bpy.ops.wm.link_append", it then proceeded to the error in question.)

When I manually added the files that you guys suggested into their appropriate places, it worked. Thank goodness you posted links to them, and they still work! Otherwise, I'm not sure I could have figured out the issue so easily.

**However, Hiero does work** whenever I add a default text object to the scene with what I assume to be the default Blender font (Bfont). The little black box pops up and everything. It even makes the respective .fntx and .png files for Bfont. The profiler, however, still crashes in this situation!

In this situation though, I can actually force the profiler to work by creating a text object using the font BDXFontMono.ttf that I manually added into the android/assets/bdx/fonts/ folder... Doing so seems to trigger Hiero to create the correct .fntx and .png files for BDXFontMono.

In short, there still seems to be something going on by default with generating the BDXFontMono files. (And there's something up with exprun.py.......)

**Comment by SolarLune** [comment]
Hey, what version of BDX are you running? And what version of Blender?

Also, what kind of computer and graphics card are you using?

**Comment by billzo** [comment]
BDX -- should be the latest release version -- 0.2.4. Unless I made a mistake and downloaded the wrong one, don't think so though. Any way to check for sure in Blender?

Blender -- Blender 2.77a x86_64 on Linux Mint

Computer --  a MacBook Pro 2011, running Linux via rEFInd...
Graphics card -- whatever graphics card comes with that, not entirely sure..

**Comment by rafcolson** [comment]
I'll have another look at this tomorrow.
